### PR TITLE
CentOS: remove stale interfaces

### PR DESCRIPTION
**Pull request by mmalchuk**
This change allow to remove all interfaces provided in
interfaces_workaround_centos_remove variable.

Signed-off-by: Maksim Malchuk <maksim.malchuk@gmail.com>

### CODE REVIEW

File: tasks/redhat.yml [hunk 1]
```diff
@@ -19,13 +19,14 @@
     path: "/etc/sysconfig/network-scripts/ifcfg-{{ item }}"
     state: absent
   when:
-    - item not in ansible_facts.interfaces
```
Comment (markgoddard) [inline tasks/redhat.yml:22]: I have seen a case where this was useful. Cloud image provided an invalid ens3. I had ens3, but did not want to configure it explicitly - only VLAN interfaces.

Comment (mmalchuk) [reply tasks/redhat.yml:22]: the simple production case: network pxe (native VLAN), network mgmt (VLAN11) - provision creates the ifcfg-ens3 without IP and ifcfg-ens3.11 with DHCP IP from ip-allocation. controllers/compute configuration have the bond0 with ens3 and ens4 interfaces, and the same management VLAN11 and IP. overcloud host configure creates ifcfg-ens3 and ifcfg-ens4 as slaves, ifcfg-bond0 and ifcfg-bond0.11 with management IP. BUT there are stale ifcfg-ens3.11 still exists on filesystem with DHCP and second default route via it.

Comment (mmalchuk) [reply tasks/redhat.yml:22]: there is the same issue with systemd-networkd configuration on Ubuntu. the solution will provide in the gerrit review later.

File: tasks/redhat.yml [hunk 2]
```diff
@@ -19,13 +19,14 @@
...
     - item not in interfaces_ether_interfaces | map(attribute='device') | list
     - item not in interfaces_bridge_interfaces | map(attribute='device') | list
     - item not in interfaces_bridge_interfaces | map(attribute='ports') | flatten | list
     - item not in interfaces_bond_interfaces | map(attribute='device') | list
     - item not in interfaces_bond_interfaces | map(attribute='bond_slaves') | flatten | list
   with_items: "{{ interfaces_workaround_centos_remove }}"
+  notify:
+    - Bounce network devices
```
[the remark this answers is not in the document]
Comment (mmalchuk) [reply tasks/redhat.yml:29]: as described above in this case this needs to not exclude stale interfaces from the scope.

[the remark this answers is not in the document]
Comment (mmalchuk) [reply tasks/redhat.yml]: Since PR#120 merged bounce not needed now. Uploaded new change.

File: tasks/redhat.yml (hunk 1 above)
Comment (markgoddard) [reply tasks/redhat.yml:22]: I have seen that issue before, and this is a reasonable solution for it.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**